packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -0,0 +1,500 @@
+           fontStyle == typedOther.fontStyle &&
+           height == typedOther.height &&
+           leading == typedOther.leading &&
+           forceStrutHeight == typedOther.forceStrutHeight;
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      fontFamily,
+      fontSize,
+      fontWeight,
+      fontStyle,
+      height,
+      leading,
+      forceStrutHeight,
+    );
+  }
+
+  /// Adds all properties prefixing property names with the optional `prefix`.
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties, { String prefix = '' }) {
+    super.debugFillProperties(properties);
+    if (debugLabel != null)
+      properties.add(MessageProperty('${prefix}debugLabel', debugLabel));
+    final List<DiagnosticsNode> styles = <DiagnosticsNode>[];
+    styles.add(StringProperty('${prefix}family', fontFamily, defaultValue: null, quoted: false));
+    styles.add(IterableProperty<String>('${prefix}familyFallback', fontFamilyFallback, defaultValue: null));
+    styles.add(DoubleProperty('${prefix}size', fontSize, defaultValue: null));
+    String weightDescription;
+    if (fontWeight != null) {
+      switch (fontWeight) {
+        case FontWeight.w100:
+          weightDescription = '100';
+          break;
+        case FontWeight.w200:
+          weightDescription = '200';
+          break;
+        case FontWeight.w300:
+          weightDescription = '300';
+          break;
+        case FontWeight.w400:
+          weightDescription = '400';
+          break;
+        case FontWeight.w500:
+          weightDescription = '500';
+          break;
+        case FontWeight.w600:
+          weightDescription = '600';
+          break;
+        case FontWeight.w700:
+          weightDescription = '700';
+          break;
+        case FontWeight.w800:
+          weightDescription = '800';
+          break;
+        case FontWeight.w900:
+          weightDescription = '900';
+          break;
+      }
+    }
+    // TODO(jacobr): switch this to use enumProperty which will either cause the
+    // weight description to change to w600 from 600 or require existing
+    // enumProperty to handle this special case.
+    styles.add(DiagnosticsProperty<FontWeight>(
+      '${prefix}weight',
+      fontWeight,
+      description: weightDescription,
+      defaultValue: null,
+    ));
+    styles.add(EnumProperty<FontStyle>('${prefix}style', fontStyle, defaultValue: null));
+    styles.add(DoubleProperty('${prefix}height', height, unit: 'x', defaultValue: null));
+    styles.add(FlagProperty('${prefix}forceStrutHeight', value: forceStrutHeight, defaultValue: null));
+
+    final bool styleSpecified = styles.any((DiagnosticsNode n) => !n.isFiltered(DiagnosticLevel.info));
+    styles.forEach(properties.add);
+
+    if (!styleSpecified)
+      properties.add(FlagProperty('forceStrutHeight', value: forceStrutHeight, ifTrue: '$prefix<strut height forced>', ifFalse: '$prefix<strut height normal>'));
+  }
+}
```

Comment:
there's a lot of code duplication in this file (e.g. the switch for font weights), which is sad. maybe a private mixin or something could make this better?

Reply:
The whole switch could be replaced by: `weightDescription = '${fontWeight.index + 1}00';`